### ISSUE TITLE
HarvestPackage doesn't work on case-sensitive filesystems due to lowercased NuGet cache

### ISSUE THREAD

**Issue by dagood** [dotnet/arcade#1655]
I'm working on getting CoreFX to build on Linux with `-allconfigurations` (https://github.com/dotnet/source-build/issues/210), and hit an error when this code tries to look for e.g. `System.Composition.Convention` in the package cache when it's really `system.composition.convention`:

https://github.com/dotnet/arcade/blob/e2ff52c754f8442cb939317d8ec72b3bb78269fc/src/Microsoft.DotNet.Build.Tasks.Packaging/src/HarvestPackage.cs#L148-L162

I'm able to work around this by changing [`Packaging.targets`](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.targets):

```diff
+    <PropertyGroup>
+      <IdToLower>$(Id.ToLowerInvariant())</IdToLower>
+    </PropertyGroup>
+
     <!-- Harvest files from old package and determine support using both runtime and additional packages -->
-    <HarvestPackage PackageId="$(Id)"
+    <HarvestPackage PackageId="$(IdToLower)"
                     PackageVersion="$(HarvestVersion)"
```

Luckily the task doesn't seem to need the actual ID, so that works for me so far.

---

However, IMO a better fix would be to make the task itself understand a little better how NuGet lays out the cache--add `ToLowerInvariant` where it's needed to find the right dirs. This would avoid a misleading variable name and let `HarvestPackage` know the package's true ID if that ends up being useful.

---

/cc @ericstj @joperezr

**Comment by joperezr** [comment]
I agree with you that the right fix should go into the task and have that be case-insensitive so that it works cross-plat.

**Comment by dagood** [comment]
I'm going to reopen this to track someone doing a proper fix rather than a workaround, but I don't have any functional objections to closing it again due to priorities.
